src/PostgREST/App.hs

```diff
@@ -206,8 +204,13 @@ app dbStructure conf apiRequest =
         else return notFound
 
     (ActionRead, TargetRoot, Nothing) -> do
-      body <- encode <$> H.query schema accessibleTables
-      return $ responseLBS status200 [jsonH] $ cs body
+      let encodeApi ti = encodeOpenAPI ti host port
```

Comment:
I did some thought on this. If we were introducing something like x-openapi-uri in the proxy to take precedence over that host / port. It is actually the same as:
 location /api/v1 {
        proxy_pass http://Host:Port/;
 }